### DISCUSSION TITLE
Mainnet/Testnet detection

### DISCUSSION THREAD

**Discussion by bakon11** [CardanoSolutions/kupo#17] [Q&A] [answered]
I am currently connecting Kupo to Ogmios. If Kupo restarts and the Ogmios instance switches from Mainnet to Testnet and vice versa.

Will Kupo continue where it left off for its respective chain net?

Thanks.

**Comment by KtorZ** [comment, marked as answer]
There's currently no network check / safeguard when connecting to Ogmios (which is arguably wrong 🤔).

In case Kupo loses the connection, it should attempt to retry to connect until it succeeds. Yet, it'll restart synchronization from "scratch", or from the last point known when the application was started. That's wrong and will need a fix.

**Comment by KtorZ** [comment]
Note that _"Yet, it'll restart synchronization from "scratch", or from the last point known when the application was started."_ is fixed by 063eb1a652a5c2ace0350e8cbfe92a5dbbb6a1a1. 

Will issue a patch release soon.

**Reply by bakon11** [reply]
Fantastic. So kupo keeps the DB separate between mainnet and testnet correct?

**Reply by KtorZ** [reply]
Kupo asks you for a working directory as for where to put the database (thereby allowing you to specify different folders). If you use the in memory database however, there's a unique one, and switching the source (I.e. the node) from testnet to mainnet (or vice versa) isn't, and won't be, supported!

**Reply by bakon11** [reply]
Cool not a problem, exactly the answer I was looking for.

Just wanted to make sure I don't duplicate work hehe.

Thanks.
